.github/workflows/gh-deployment.yml

```diff
@@ -0,0 +1,39 @@
+name: Deployment to GitHub Pages
+
+permissions:
+  contents: write
+
+on:
+  # call this workflow when the workflow Main Validation is completed after a push to main
+  workflow_run:
+    workflows:
+      - 'Continuous Integration'
+    branches:
+      - main
+    types:
+      - completed
+
+jobs:
+  deploy:
+    if: ${{ github.event.workflow_run.conclusion == 'success' }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Begin CI...
+        uses: actions/checkout@v3
+
+      - name: Set up Node
+        uses: actions/setup-node@v3
+        with:
+          node-version-file: .node-version
+
+      - name: Install dependencies
+        run: npm install
```

Comment:
```suggestion
        run: pnpm install
```

Reply:
you may need to use this action to install pnpm before using it 😊  https://github.com/pnpm/action-setup